ZFSin/zfs/module/zfs/zfs_vnops_windows.c

```diff
@@ -4274,7 +4276,9 @@ diskDispatcher(
 		switch (IrpSp->MinorFunction) {
 		case IRP_MN_MOUNT_VOLUME:
 			dprintf("IRP_MN_MOUNT_VOLUME disk\n");
+			FsRtlEnterFileSystem();
 			Status = zfs_vnop_mount(DeviceObject, Irp, IrpSp);
+			FsRtlExitFileSystem();
 			break;
 		case IRP_MN_USER_FS_REQUEST:
 			dprintf("IRP_MN_USER_FS_REQUEST: FsControlCode 0x%x\n",
```

Comment:
IRP_MN_USER_FS_REQUEST also needs the FsRtlEnterFileSystem() call so it is better to use that call like this (and sam ein the ioctlDispatcher too).
case IRP_MJ_FILE_SYSTEM_CONTROL:
{
    FsRtlEnterFileSystem();
		switch (IrpSp->MinorFunction) {
		case IRP_MN_MOUNT_VOLUME:
			dprintf("IRP_MN_MOUNT_VOLUME disk\n");
			Status = zfs_vnop_mount(DeviceObject, Irp, IrpSp);
			break;
		case IRP_MN_USER_FS_REQUEST:
			dprintf("IRP_MN_USER_FS_REQUEST: FsControlCode 0x%x\n",
				IrpSp->Parameters.FileSystemControl.FsControlCode);
			Status = user_fs_request(DeviceObject, Irp, IrpSp);
			break;
		}
    FsRtlExitFileSystem();
 }
break;